### DISCUSSION TITLE
How the Action Priority List works?

### DISCUSSION THREAD

**Discussion by marekCs** [simulationcraft/simc#6168] [General]
Hello, could someone please explain to me how the Action Priority List works? Respectively what principles and in what order to maintain? I mean individual sections, e.g. for Destruction Warlock:
1) actions.precombat - I understand this is executed before combat.
2) Default action list Executed every time the actor is available - this I understand, whenever one condition is met, execute.
3) actions.cds - this means what? I thought cooldowns were already resolved in the previous section? Then I still had a section in an older version of the program: 
4) actions.gosup_infernal - I also assumed this was already covered in point 2. How is default rotation different from cds and gosup_infernal? Which takes priority, or does it overlap?

Then I have a subquestion, are there explanations of the variable names anywhere? For example, this:

rain_of_fire,if=pet.infernal.active&(buff.crashing_chaos.down|!talent.grimoire_of_supremacy.enabled)&(!cooldown.havoc.ready|active_enemies>3)

pet.Infernal.active means that Infernal is visible or that it is not on cooldown?
If it is visible then I don't understand the next condition buff.crashing_chaos.down, because buff crashing_chaos is only when infernal is active (visible) for 20 seconds, so is this condition unnecessary?

**Comment by marekCs** [comment]
Solved, I have found this: https://github.com/simulationcraft/simc/wiki/ :)

**Comment by marekCs** [comment]
ok still don't know what is the ability list "actions.gosup_infernal" means its when the pet is down and active? And also dont know what it does mean this: immolate,if=talent.internal_combustion.enabled&remains<duration*0.5

the "remains" and "duration! option, that belongs to what spell, specifically? A bit confused as the basic specification is missing.

**Reply by Azevara** [reply]
If a spell is not specified, things like "remains" refer to the spell in the action, I believe this is outlined on the wiki. Also, as you mentioned before, the specific action list you are asking about does not exist any longer.

**Reply by marekCs** [reply]
Hello thank you for your answer. Yes, then I also figured it would probably refer to the Immolate spell, but thanks for the confirmation. As for it no longer existing, I'm playing wow version of 8.3.7 on privátně server and using simcraft version 8.3.02 (I couldn't find a corresponding simcraft version for wow 8.3.7).

**Comment by marekCs** [comment]
And how can I please understand this line? **prev_gcd.1.aspect_of_the_wild**   

all the lines"
<html>
<body>


Default action list Executed every time the actor is available.
--

E | 1.00 | auto_shot
  | 0.00 | use_items,if=prev_gcd.1.aspect_of_the_wild\|target.time_to_die<20


</body>
</html>

---------------------------------------------
That is, if "Aspect of the Wild" has already been used use all the list below?
